src/psyclone/psyGen.py

```diff
@@ -4748,28 +4750,50 @@ def _parse_dimensions(dimensions):
         from fparser.two.utils import walk_ast
         from fparser.two import Fortran2003
         shape = []
-        for dim in walk_ast(dimensions.items, [Fortran2003.Assumed_Shape_Spec,
-                                               Fortran2003.Explicit_Shape_Spec,
-                                               Fortran2003.Assumed_Size_Spec]):
+
+        if isinstance(dimensions, (Fortran2003.Assumed_Shape_Spec,
+                                   Fortran2003.Explicit_Shape_Spec,
+                                   Fortran2003.Assumed_Size_Spec)):
+            dimensions_list = [dimensions]
+        else:
+            # Depth-first-search list of shape specs
+            dimensions_list = walk_ast(
+                dimensions.items,
```

Comment:
If you replace `dimensions.items` with `[dimensions]` then could you remove the earlier `if` - or would that return a scalar rather than a list?

Reply:
True, much better :)